install.sh

```diff
@@ -0,0 +1,437 @@
+#!/bin/sh
+# shellcheck shell=dash
+# shellcheck disable=SC2039
+
+# This is just a little script that can be downloaded from the internet to install Scarb.
+# It just does platform detection, downloads the release archive, extracts it and tries to make
+# the `scarb` binary available in $PATH in least invasive way possible.
+#
+# It runs on Unix shells like {a,ba,da,k,z}sh. It uses the common `local` extension.
+# Note: Most shells limit `local` to 1 var per line, contra bash.
+#
+# Most of this code is based on/copy-pasted from rustup and protostar installers.
+
+set -u
+
+SCARB_REPO="https://github.com/software-mansion/scarb"
+INSTALL_ROOT="${HOME}/.local/scarb"
+LOCAL_BIN="${HOME}/.local/bin"
+
+usage() {
+  cat <<EOF
+The installer for Scarb
+
+Usage: install.sh [OPTIONS]
+
+Options:
+    -v        Specify Scarb _requested_version to install
+    -h        Print help information
+
+For more information, check out https://docs.swmansion.com/scarb/download.
+EOF
+}
+
+main() {
+  need_cmd chmod
+  need_cmd curl
+  need_cmd grep
+  need_cmd mkdir
+  need_cmd mktemp
+  need_cmd rm
+  need_cmd sed
+  need_cmd tar
+  need_cmd uname
+
+  local _requested_ref="latest"
+  local _requested_version="latest"
+  while getopts ":hv:" opt; do
+    case $opt in
+    h)
+      usage
+      exit 0
+      ;;
+    v)
+      _requested_ref="tag/v${OPTARG}"
+      _requested_version="$OPTARG"
+      ;;
+    \?)
+      err "invalid option -$OPTARG"
+      ;;
+    :)
+      err "option -$OPTARG requires an argument"
+      ;;
+    esac
+  done
+
+  resolve_version "$_requested_version" "$_requested_ref" || return 1
+  local _resolved_version=$RETVAL
+  assert_nz "$_resolved_version" "resolved_version"
+
+  get_architecture || return 1
+  local _arch="$RETVAL"
+  assert_nz "$_arch" "arch"
+
+  local _tempdir
+  if ! _tempdir="$(ensure mktemp -d)"; then
+    # Because the previous command ran in a subshell, we must manually propagate exit status.
+    exit 1
+  fi
+
+  ensure mkdir -p "$_tempdir"
+
+  create_install_dir "$_requested_version"
+  local _installdir=$RETVAL
+  assert_nz "$_installdir" "installdir"
+
+  download "$_resolved_version" "$_arch" "$_installdir" "$_tempdir"
+
+  say "installed scarb to ${_installdir}"
+
+  create_symlink "$_installdir"
+
+  echo
+  if [[ ":$PATH:" != *":${LOCAL_BIN}:"* ]]; then
+    add_local_bin_to_path
+    echo "Then, run 'scarb --version' to verify your installation. Happy coding!"
+  else
+    echo "Scarb has been successfully installed and should be already available in your PATH."
+    echo "Run 'scarb --version' to verify your installation. Happy coding!"
+  fi
+
+  local _retval=$?
```

Comment:
This will always be 0, because it directly follows `echo` command

Reply:
Done